tools/azure-sdk-tools/devtools_testutils/azure_testcase.py

```diff
@@ -182,34 +187,52 @@ def setUp(self):
     def tearDown(self):
         return super(AzureTestCase, self).tearDown()
 
-    def create_basic_client(self, client_class, **kwargs):
 
-        tenant_id = os.environ.get("AZURE_TENANT_ID", None)
-        client_id = os.environ.get("AZURE_CLIENT_ID", None)
-        secret = os.environ.get("AZURE_CLIENT_SECRET", None)
 
-        if tenant_id and client_id and secret and self.is_live:
+    def _create_credential(self, client_secret_credential):
+        return client_secret_credential(
+            tenant_id=self.tenant_id,
+            client_id=self.client_id,
+            client_secret=self.secret
+        )
+
+    def _get_real_credential(self, is_async):
+        from azure.identity import ClientSecretCredential
+        if is_async:
+            from azure.identity.aio import ClientSecretCredential
+        return self._create_credential(ClientSecretCredential)
+
+    def _get_fake_credential(self, is_async):
```

Comment:
depending on the logic, I tend to merge this into `_get_real_credential` to be:

`_get_credential(self, is_async=False, is_real=True, ...)`, more of using parameters to control the behavior instead of having two separate methods.

well, that's my personal preference based on what I learn from our libraries and python :)

Reply:
As long as it's private I don't mind, though too much boolean ends up being `method(True, False, True, False, False)` :p

Reply:
thanks @seankane-msft , I see you're reusing `self._is_live` for differentiate fake from real, a smart move